src/hls-adapter.js

```diff
@@ -539,8 +539,8 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
   load(startTime: ?number): Promise<Object> {
     if (!this._loadPromise) {
       this._startTime = startTime;
-      this._loadPromise = new Promise(resolve => {
-        this._resolveLoad = resolve;
+      this._loadPromise = new Promise((resolve, reject) => {
+        this._loadPromiseHandlers = {resolve, reject};
```

Comment:
declare it like the others and reset it where need

Reply:
need to be reset only on resolve and reject (2 places)

Reply:
in _reset and destroy also no?

Reply:
one of the two options (manifest loaded , manifest loading error) will always have to happen, no third option, and it is reset to null in both these cases, so i think any additional reset is redundant